### ISSUE TITLE
Add chflags on FUSE3

### ISSUE THREAD

**Issue by sganis** [billziss-gh/winfsp#286]
Please implement chflags operation in Fuse3 API.

**Comment by billziss-gh** [comment]
@sganis thanks for the enhancement request.

Implementing this should be relatively easy as we would just wrap the equivalent operation in FUSE2. The bigger issue is that this would introduce a change that is incompatible to FUSE3 as implemented in libfuse. This is the primary reason why I have not implemented this so far.

**Comment by mheinzler** [comment]
Hey @billziss-gh,
first of all thanks for this great tool. I'm currently working on a C++ filesystem that also grew into a kind of wrapper around libfuse and WinFsp and now found myself also missing this operation.

Do I understand correctly that the main issue for extending `fuse3_operations` would be breaking future compatibility with libfuse's `fuse_operations`? If yes, what would you think about adding something like `fuse3_operations_ex` with some additional ops and extending the API entry points to forward these to the FUSE2 layer? Something like this could certainly also be useful to expose more Windows specifics while using the FUSE API.

I would be able to work on a PR for this.

**Comment by billziss-gh** [comment]
@mheinzler 

> Do I understand correctly that the main issue for extending fuse3_operations would be breaking future compatibility with libfuse's fuse_operations?

You are correct. I would not mind adding `chflags`, but I would want to do so in coordination with libfuse. There are other features that FUSE2 has that I would also want to see in FUSE3 (e.g. case-insensitive file systems), but I hesitate to add them as to not break compatibility.

Can you explain a bit more about how you are seeing the `fuse3_operations_ex` work?

**Comment by mheinzler** [comment]
Here's some basic code that might explain it better (using a `winfsp` prefix to make it clear that this has nothing to do with libfuse):
```c
struct winfsp_fuse_operations {
  // Windows-specific extensions (maybe even config options?)
  int (*chflags)(const char *path, uint32_t flags);
};

const fuse_operations fuse_ops = {
  getattr, readlink, ... 
};

const winfsp_fuse_operations win_ops = {
  my_chflags, ...
};

winfsp_fuse_main(argc, argv, &fuse_ops, &win_ops, data);
```

Something like this. I understand that it is not nice to have a kind of shadow API in parallel to the official one, but I would guess libfuse would never add support for Windows-only stuff.

On the user side it would be pretty easy to implement with just a few conditional macros. That way a filesystem could provide better integration into Windows while still sharing most of the code with libfuse.
